src/client/app/graph/arrow/arrow.component.ts

```diff
@@ -78,4 +89,70 @@ export class ArrowComponent implements OnInit {
     return pointsStr;
   }
 
+  onMousedown(event: MouseEvent) {
+    if (EventUtils.eventIsFromPrimaryButton(event)) {
+      this.tipDragging = true;
+      const canvasBounds = this.canvasBoundsGetter();
+      this.lastDragPnt = {
+        x: event.clientX - canvasBounds.left,
+        y: event.clientY - canvasBounds.top
+      };
+    }
+  }
+
+  // Put mousemove on document to allow dragging outside of canvas
+  //noinspection JSUnusedGlobalSymbols
+  @HostListener('document:mousemove', ['$event'])
+  onMousemove(event: MouseEvent) {
+    if (this.tipDragging) {
+      event.preventDefault();
+      const canvasBounds = this.canvasBoundsGetter();
+      const newDragPnt = {
+        x: event.clientX - canvasBounds.left,
+        y: event.clientY - canvasBounds.top
+      };
+      this.tipPosition.x += newDragPnt.x - this.lastDragPnt.x;
+      this.tipPosition.y += newDragPnt.y - this.lastDragPnt.y;
+      const newTipPosition =
+        this.canvasService.viewportCoordToCanvasCoord(this.tipPosition);
+      this.arrow.tipPosition = {x: newTipPosition.x, y: newTipPosition.y};
```

Comment:
Style these inline objects to match what you have above? (for readability, [96], [110])

Reply:
Done